plugin/src/main/java/org/openrewrite/gradle/RewritePlugin.java

```diff
@@ -190,12 +168,20 @@ private static void configureProject(Project project, RewriteExtension extension
         });
     }
 
-    private Set<File> getResolvedDependencies(Configuration rewriteConf) {
-        if (resolvedDependencies != null) {
-            return resolvedDependencies;
-        }
+    private Set<File> getResolvedDependencies(Project project, RewriteExtension extension, Configuration rewriteConf) {
+        if (resolvedDependencies == null) {
+            Dependency[] dependencies = Stream.concat(
+                    knownRewriteDependencies(extension, project.getDependencies()).stream(),
+                    rewriteConf.getDependencies().stream()
+            ).toArray(Dependency[]::new);
+            // By using a detached configuration, we separate this dependency resolution from the rest of the project's
+            // configuration. This also means that Gradle has no criteria with which to select between variants of
+            // dependencies which expose differing capabilities. So those must be manually configured
+            Configuration detachedConf = project.getConfigurations().detachedConfiguration(dependencies);
```

Comment:
Would it be possible to add a flag to the extension to disable the detached configurations behavior?

Reply:
Potentially? The unfortunate part of doing that would mean moving further away from being compatible with the configuration cache feature (all tasks are marked as not compatible presently).

Reply:
> Would it be possible to add a flag to the extension to disable the detached configurations behavior?
> 
> Or detect the spring plugin and use only use the detached configuration if it's present?

I will check

Reply:
I add a Gradle parameter so you can add `-Prewrite.detachResolve` to detach the resolver.

Reply:
@arlaneenalra, I'm trying to wrap my head around this, but is there a reason for your all's plugin to try to bother with central dependency version management? It seems like the `rewrite` configuration not being a detached configuration in all cases is causing some pretty wide spread problems.

The `rewrite` configuration is also really meant to be isolated just for the purposes of this plugin, so other plugins manipulating it's versions has shown to be detrimental (case in point from @nicolasb29's issue).

I'm really wondering if just going back to always being a detached configuration may be the right thing to do rather than trying to have a toggle to go one way or the other...

Reply:
>@arlaneenalra, I'm trying to wrap my head around this, but is there a reason for your all's plugin to try to bother with central dependency version management for the rewrite configuration specifically? It seems like the rewrite configuration not being a detached configuration in all cases is causing some pretty wide spread problems.

For our use case, we have a local publishing process that redirects some dependencies when doing development. That process doesn't work for an detached configuration like this without special handling (i.e. a custom configuration class). Our work arounds wound up injecting conflicting dependency jars and causing random failures because of the conflicts.  If this is really not viable, I'll have to do some more digging to see if there are other alternatives .. hmm

Reply:
For what it's worth, what I typically do is use Gradle's `publishToMavenLocal`, then bring in the local snapshot jar via Gradle's native dependency resolution.